src/system/unwind.c

```diff
@@ -155,6 +167,21 @@ void report_data_abort(uint32_t _sp) {
 	fprintf(stderr, "\t%p\n", (void*)vrs.core.r[R_PC]);
 	__gnu_Unwind_Backtrace(trace_fn, NULL, &vrs);
```

Comment:
It would also make more sense for the trace to not be a global variable. Instead you can create the struct locally here and pass the pointer to the struct instead of NULL here. and then you can use it as d in the trace_fn.